### PR TITLE
Add a study view warning

### PR DESCRIPTION
**Pull request by Leowisd**
Fix https://github.com/cBioPortal/cbioportal/issues/4065

Default value will be removed when it comes from backend

### CODE REVIEW

File: src/config/serverConfigDefaults.ts [hunk 1]
```diff
@@ -45,6 +45,8 @@ const ServerConfigDefaults: Partial<IServerConfig> = {
     skin_description:
         'The cBioPortal for Cancer Genomics provides visualization, analysis and download of large-scale cancer genomics data sets',
     show_genomenexus: true,
+    skin_study_view_blurb:
```
Comment (zhx828) [inline src/config/serverConfigDefaults.ts:48]: This default will be removed before merging. The value should come from the server which https://github.com/cBioPortal/cbioportal/pull/7718 is trying to address that.

**Comment by alisman** [comment]
@Leowisd could you guys put a screenshot in here so i can see appearance of warning?

**Comment by Leowisd** [comment]
Like this
image

**Comment by alisman** [comment]
@jjgao @zhx828 do we want to do a little extra work to make this warning dismissable (stored in localstorage)

**Comment by alisman** [comment]
@Leowisd the spacing around that alert isn't great.   lets wait to see what JJ says about dismissability.  if we user can dismiss warning, then it could be above the study and that would ease positioning.

**Comment by inodb** [comment]
Not sure if the warning should be this big/visible, what do you think @zhx828 @jjgao ? Another option would be to append the text to the description of the study. E.g. similar to https://cbioportal.mskcc.org/study/summary?id=mskimpact. There is some red text warning

**Comment by zhx828** [comment]
> Not sure if the warning should be this big/visible, what do you think @zhx828 @jjgao ? Another option would be to append the text to the description of the study. E.g. similar to https://cbioportal.mskcc.org/study/summary?id=mskimpact. There is some red text warning

Good point, I believe that red text is very similar to the message we are about to add. I will defer the decision to the @cBioPortal/product team.

**Comment by Leowisd** [comment]
So if like this
image

**Comment by alisman** [comment]
@inodb @zhx828 i would be in favor of a dismisseable warning.  That we can get people to notice without taking up real estate permanently.

**Comment by schultzn** [comment]
But then they will dismiss the warning and forget about it, and use the URL in a paper again. I am not sure that we have a better choice here.

**Comment by Leowisd** [comment]
image
image
image

File: src/appShell/App/Container.tsx [hunk 2]
```diff
@@ -90,6 +91,14 @@ export default class Container extends React.Component<IContainerProps, {}> {
 
                 <div className="pageTopContainer">
                     <UserMessager />
+                    {localStorage.getItem('warning_dismiss') !== 'Yes' &&
```
Comment (alisman) [inline src/appShell/App/Container.tsx:94]: that key should be more specific to this particular case.   'private_study_link_warning'

File: src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        super(props);
+
+        this.handleHide = this.handleHide.bind(this);
+        this.handleShow = this.handleShow.bind(this);
+        this.handleDismiss = this.handleDismiss.bind(this);
+        this.handleModalShow = this.handleModalShow.bind(this);
+        this.handleModalHide = this.handleModalHide.bind(this);
+        this.handleChecked = this.handleChecked.bind(this);
+
+        this.state = {
```
Comment (alisman) [inline src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx:29]: we don't use React state.  We use mobx observables.   Not a big deal but you should familiarize yourself with how it works.   Look at any other component and you will see see @observable decorator on instance properties.  And you'll see @observer decorator on the class.  This wires up render to observe changes to observable properties exactly as setState does.

File: src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+export default class StudyViewWarning extends React.Component<
+    IStudyViewWarningBannerProps,
+    IStudyViewWarningBannerStates
+> {
+    constructor(props: any) {
+        super(props);
+
+        this.handleHide = this.handleHide.bind(this);
```
Comment (alisman) [inline src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx:22]: you can do this with @autobind decorator

Comment (alisman) [reply src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx:22]: that way, you don't have to bother with this stuff in constructor

**Pull request closed without merging**